api/tests/routes/backoffice/admin_test.py

```diff
@@ -103,6 +106,99 @@ def test_update_role_with_empty_permissions(self, authenticated_client):
+        permission2 = perm_factories.PermissionFactory(name=perm_models.Permissions.READ_ADMIN_ACCOUNTS.value)
+        permission3 = perm_factories.PermissionFactory(name=perm_models.Permissions.FEATURE_FLIPPING.value)
+        perms = [permission1, permission2, permission3]
+        perm_factories.PermissionFactory(name=perm_models.Permissions.SUSPEND_USER.value)
+        old_perms = [permission1, permission2]
+        role_to_edit = perm_factories.RoleFactory(
+            name=perm_models.Roles.SUPPORT_N1.name,
+            permissions=old_perms,
+        )
+        role_not_to_edit = perm_factories.RoleFactory(permissions=old_perms)
+
+        new_perms = [permission2, permission3]
+        base_form = {}
+        for perm in perms:
+            base_form[perm.name] = perm in new_perms
+
+        response = self.post_to_endpoint(authenticated_client, role_id=role_to_edit.id, form=base_form)
+        assert response.status_code == 303
+
+        action = history_models.ActionHistory.query.one()
+        assert action.actionType == history_models.ActionType.ROLE_PERMISSIONS_CHANGED
+        assert action.authorUserId == legit_user.id
+        assert action.extraData == {
+            "role_name": perm_models.Roles.SUPPORT_N1.value,
+            "modified_info": {
+                role_not_to_edit.permissions[0].name: {"new_info": False, "old_info": True},
+                role_to_edit.permissions[1].name: {"new_info": True, "old_info": False},
+            },
+        }
+
+    def test_log_update_role_no_modification(self, legit_user, authenticated_client):
+        permission1 = perm_factories.PermissionFactory(name=perm_models.Permissions.MANAGE_PERMISSIONS.value)
+        permission2 = perm_factories.PermissionFactory(name=perm_models.Permissions.READ_ADMIN_ACCOUNTS.value)
+        perm_factories.PermissionFactory(name=perm_models.Permissions.SUSPEND_USER.value)
+        old_perms = [permission1, permission2]
+        role_to_edit = perm_factories.RoleFactory(
+            name=perm_models.Roles.SUPPORT_N1.name,
+            permissions=old_perms,
+        )
+        base_form = {}
+        for perm in old_perms:
+            base_form[perm.name] = perm in old_perms
+
+        response = self.post_to_endpoint(authenticated_client, role_id=role_to_edit.id, form=base_form)
+        assert response.status_code == 303
+
+        history_rows = html_parser.extract_table_rows(response.data)
+
+        assert len(history_rows) == 0
+
+
+class GetRolesHistoryTest(GetEndpointHelper):
+    endpoint = "backoffice_web.get_roles_history"
+    needed_permission = perm_models.Permissions.MANAGE_PERMISSIONS
+    expected_num_queries = 3
+
+    def test_get_log_history_admin(self, legit_user, authenticated_client):
+        permission1 = perm_factories.PermissionFactory(name=perm_models.Permissions.MANAGE_PERMISSIONS.value)
+        permission2 = perm_factories.PermissionFactory(name=perm_models.Permissions.READ_ADMIN_ACCOUNTS.value)
```

Comment:
Non, c'est `name` dans `name`, il faut que le test soit cohérent avec ce qui se passe dans la réalité !
```suggestion
        permission1 = perm_factories.PermissionFactory(name=perm_models.Permissions.MANAGE_PERMISSIONS.name)
        permission2 = perm_factories.PermissionFactory(name=perm_models.Permissions.READ_ADMIN_ACCOUNTS.name)
```

Reply:
raison qui m'empeche de le faire (aussi sur l'autre test): tests/routes/backoffice/admin_test.py::GetRolesHistoryTest::test_get_log_history_admin - sqlalchemy.exc.IntegrityError: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "permission_name_key"

investigation